tests/python/conftest.py

```diff
@@ -47,3 +50,186 @@ def unload_openassetio_modules():
     # Restore the previously imported modules
     for name, module in to_delete.items():
         sys.modules[name] = module
+
+
+@pytest.fixture
+def mock_manager_interface():
+    """
+    Fixture for a `ManagerInterface` that asserts parameter types and
+    forwards method calls to an internal public `mock.Mock` instance.
+    """
+    return ValidatingMockManagerInterface()
+
+
+class ValidatingMockManagerInterface(ManagerInterface):
+    """
+    `ManagerInterface` implementation that asserts parameter types then
+    calls an internal mock.
+
+    All method calls first have their parameter types validated, then
+    are forwarded to an internal `mock.Mock` instance, which is
+    `autospec`ed to `ManagerInterface` and accessible as a public `mock`
+    member variable, so that expectations can be configured.
+
+    @see mock_manager_interface
+    """
+
+    # pylint: disable=too-many-public-methods,too-many-arguments
+
+    def __init__(self):
+        super(ManagerInterface, self).__init__()
```

Comment:
[fix] I _think_ this is now just `super().__init__()`

https://peps.python.org/pep-3135/

Reply:
Perhaps it's time to remove pylint disable config from `pyproject.toml` where we had
```
    # Disable Python 3 improvements:
    # - We've not definitely ruled 2 out
    # - Much of this code will move to C++
```

Reply:
Fixed for this particular case in 1317b51b

Reply:
Ahh yes, sorry I forgot about those. I'd taken them out in the new `MediaCreation` repo.  Added https://github.com/TheFoundryVisionmongers/OpenAssetIO/issues/364 to track this.